File: dotnet/src/MyDotey.SCF/Facade/ConfigurationSources.cs
using System;

namespace MyDotey.SCF.Facade
{
    /**
     * @author koqizhao
     *
     * May 17, 2018
     */
    public class ConfigurationSources
    {
        protected ConfigurationSources()
        {

        }

        public static ConfigurationSourceConfig NewConfig(String name)
        {
            return NewConfigBuilder().SetName(name).Build();
        }

        public static ConfigurationSourceConfig.IBuilder NewConfigBuilder()
        {
            return new DefaultConfigurationSourceConfig.Builder();
        }
    }
}

File: java/src/main/java/org/mydotey/scf/facade/ConfigurationSources.java
package org.mydotey.scf.facade;

import org.mydotey.scf.ConfigurationSourceConfig;
import org.mydotey.scf.DefaultConfigurationSourceConfig;

/**
 * @author koqizhao
 *
 * May 17, 2018
 */
public class ConfigurationSources {

    protected ConfigurationSources() {

    }

    public static ConfigurationSourceConfig newConfig(String name) {
        return newConfigBuilder().setName(name).build();
    }

    public static ConfigurationSourceConfig.Builder newConfigBuilder() {
        return new DefaultConfigurationSourceConfig.Builder();
    }

}
